llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -624,9 +662,9 @@ int main(int argc, char **argv) {
     errs() << "no actions specified; try --help for usage info\n";
     return 1;
   }
-  if (IROutputOnly && DoSplit) {
-    errs() << "error: -" << SplitMode.ArgStr << " can't be used with -"
-           << IROutputOnly.ArgStr << "\n";
+  if (IROutputOnly && (DoSplit && SplitMode != SPLIT_AUTO)) {
```

Comment:
Why SPLIT_AUTO is allowed with IR output only?

Reply:
> Why SPLIT_AUTO is allowed with IR output only?

Technically this is me being a bit lazy to dig further into the driver and do some changes there. Basically, there are compilation flows (related to AOT), where `sycl-post-link` is involved with `-ir-output-only` option and according to my changes, `-split=` is almost always passed to `sycl-post-link` (unless split was explicitly disabled) and it triggered errors in some LIT tests.

Reply:
OK, so the `auto` heuristics in presence of `-ir-output-only` is fixed to "don't split" - ? please add a comment then, to user-visible description too

Reply:
> OK, so the auto heuristics in presence of -ir-output-only is fixed to "don't split" - ? please add a comment then, to user-visible description too

That's correct, updated help message of the tool in edfa44cc9a08c732d48f3bfe2b8caf3f8841d4ac